reco_utils/recommender/newsrec/newsrec_utils.py

```diff
@@ -0,0 +1,291 @@
+            # attention
+            "attention_hidden_dim",
+            "loss",
+            "data_format",
+            "dropout"
+        ]
+    elif f_config["model_type"] in ['npa', 'NPA']:
+        required_parameters = [
+            "doc_size",
+            "his_size",
+            "user_num",
+            "wordEmb_file",
+            "word_size",
+            "npratio",
+            "data_format",
+            "word_emb_dim",
+            # npa
+            "user_emb_dim",
+            "filter_num",
+            "cnn_activation",
+            "window_size",
+            # attention
+            "attention_hidden_dim",
+            "loss",
+            "data_format",
+            "dropout"
+        ]
+    else:
+        required_parameters = []
+
+    # check required parameters
+    for param in required_parameters:
+        if param not in f_config:
+            raise ValueError("Parameters {0} must be set".format(param))
+
+    if f_config["model_type"] in ['nrms', 'NRMS', 'lstur', 'LSTUR']:
+        if f_config["data_format"] != "news":
+            raise ValueError(
+                "For nrms and naml model, data format must be 'news', but your set is {0}".format(
+                    f_config["data_format"]
+                )
+            )
+    elif f_config["model_type"] in ['naml', 'NAML']:
+        if f_config["data_format"] != "naml":
+            raise ValueError(
+                "For nrms and naml model, data format must be 'naml', but your set is {0}".format(
+                    f_config["data_format"]
+                )
+            )
+    
+    check_type(f_config)
+
+
+
+
+def create_hparams(flags):
+    """Create the model hyperparameters.
+
+    Args:
+        flags (dict): Dictionary with the model requirements.
+
+    Returns:
+        obj: Hyperparameter object in TF (tf.contrib.training.HParams).
+    """
+    return tf.contrib.training.HParams(
+        # data
+        data_format=flags["data_format"] if "data_format" in flags else None,
+        iterator_type=flags["iterator_type"] if "iterator_type" in flags else None,
+        # models
+        wordEmb_file=flags["wordEmb_file"] if "wordEmb_file" in flags else None,
+        doc_size=flags["doc_size"] if "doc_size" in flags else None,
+        title_size=flags["title_size"] if "title_size" in flags else None,
+        body_size=flags["body_size"] if "body_size" in flags else None,
+        word_emb_dim=flags["word_emb_dim"] if "word_emb_dim" in flags else None,
+        word_size=flags["word_size"] if "word_size" in flags else None,
+        user_num=flags["user_num"] if "user_num" in flags else None,
+        vert_num=flags["vert_num"] if "vert_num" in flags else None,
+        subvert_num=flags["subvert_num"] if "subvert_num" in flags else None,
+        his_size=flags["his_size"] if "his_size" in flags else None,
+        npratio=flags["npratio"] if "npratio" in flags else None,
+        dropout=flags["dropout"] if "dropout" in flags else 0.0,
```

Comment:
This could all be a bit cleaner using get, e.g.
```suggestion
        npratio=flags.get("npratio"),
        dropout=flags.get("dropout", 0.0),
```